libs/designer-ui/src/lib/dynamicallyaddedparameter/helper.ts

```diff
@@ -123,20 +123,45 @@ function getDescriptionForDynamicallyAddedParameterType(type: DynamicallyAddedPa
  *              required: ['text', 'text_1'],
  *          }
  *      }
- *
+ * Sample shape of schema object JSON expected by FlowRP that we are serializing/deserializing for Hybrid trigger:
+ *         schema: {
+ *           rows: {
+ *             type: 'array',
+ *             items: {
+ *               type: 'object',
+ *               properties: {
+ *                 type: 'object',
+ *                 properties: {
+ *                   'text' : ...,
+ *                   'text_1': ...,
+ *                   'number': ...,
+ *                 }
+ *                 required: ['text', 'text_1'],
+ *               },
+ *             }
+ *           }
  * @param value - valueSegment provided to us by rest of designer parent components
+ * @param isManualTrigger - Flag that sets the rootObject to required schema type.
  * @param onChange - handler to update value when the user changes their input in one of the dynamic parameters
  * @returns - array of props to render DynamicallyAddedParameter editors with
  */
-export function deserialize(value: ValueSegment[]): DynamicallyAddedParameterProps[] {
+export function deserialize(value: ValueSegment[], isManualTrigger = false): DynamicallyAddedParameterProps[] {
   if (!value || value.length === 0 || !value[0].value) {
     return [];
   }
   // ASSUMPTION: for manual trigger, we assume there is *only one* ValueSegment which contains the required data
+  // ASSUMPTION: for hybrid triggers, the floating action segmets are always nested within rows.items
   const rootObject = JSON.parse(value[0].value);
 
   const retval: DynamicallyAddedParameterProps[] = [];
-  for (const [schemaKey, propertiesUnknown] of Object.entries(rootObject?.properties)) {
+  let itemsProperties: any;
+  if (isManualTrigger) {
```

Comment:
What is the appropriate "default" behavior for this editor? This reads like you are special casing manual trigger and otherwise assume the general usage will be like for hybrid triggers.

With the context I have, I feel it should be the other way around. Like hybrid trigger gets special treatment if a flag is set. I don't think the default should be using rootObject.properties.rows.items : rootObject.rows.items to deserialize

Reply:
yes, currently working on adding a flag for hybridTriggers and leaving the default route unchanged